test/integration/listener_lds_integration_test.cc

```diff
@@ -902,9 +948,100 @@ TEST_P(ListenerFilterIntegrationTest, MixNoInspectDataFilterAndInspectDataFilter
   tcp_client->close();
 }
 
-INSTANTIATE_TEST_SUITE_P(IpVersions, ListenerFilterIntegrationTest,
-                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
-                         TestUtility::ipTestParamsToString);
+// Only update the order of listener filters, ensure the listener filters
+// was update.
+TEST_P(ListenerFilterIntegrationTest, UpdateListenerFilterOrder) {
+  // Add two listener filters. The first filter will peek 5 bytes data,
+  // the second filter will drain 2 bytes data. Expect the upstream will
+  // receive 3 bytes data.
+  config_helper_.addListenerFilter(R"EOF(
+      name: inspect_data1
+      typed_config:
+        "@type": type.googleapis.com/test.integration.filters.InspectDataListenerFilterConfig
+        max_read_bytes: 2
+        close_connection: false
+        drain: true
+        )EOF");
+  config_helper_.addListenerFilter(R"EOF(
+      name: inspect_data2
+      typed_config:
+        "@type": type.googleapis.com/test.integration.filters.InspectDataListenerFilterConfig
+        max_read_bytes: 5
+        close_connection: false
+        )EOF");
+  std::string data = "hello";
+  std::string data_after_drain = data.substr(2, std::string::npos);
+  config_helper_.addConfigModifier([](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+    // Add the static cluster to serve LDS.
+    auto* lds_cluster = bootstrap.mutable_static_resources()->add_clusters();
+    lds_cluster->MergeFrom(bootstrap.static_resources().clusters()[0]);
+    lds_cluster->set_name("lds_cluster");
+    ConfigHelper::setHttp2(*lds_cluster);
```

Comment:
TIL!

Reply:
sorry, I didn't get what you mean here.

Reply:
never mind. I didn't know this helper function. Thanks！